File: mobile/src/main/java/com/windscribe/mobile/account/AccountPresenter.java
/*
 * Copyright (c) 2021 Windscribe Limited.
 */

package com.windscribe.mobile.account;

import android.content.Context;

public interface AccountPresenter {

    void onAddEmailClicked(String tvEmailText);

    void observeUserData(AccountActivity accountActivity);

    void onCodeEntered(String code);

    void onDestroy();

    void onEditAccountClicked();

    void onResendEmail();

    void onUpgradeClicked(String textViewText);

    void onLazyloginClicked();

    void setLayoutFromApiSession();

    void setTheme(Context context);
}

File: mobile/src/main/java/com/windscribe/mobile/account/AccountPresenter.kt
/*
 * Copyright (c) 2021 Windscribe Limited.
 */
package com.windscribe.mobile.account

import android.content.Context

interface AccountPresenter {
    fun onAddEmailClicked(tvEmailText: String)
    fun observeUserData(accountActivity: AccountActivity)
    fun onCodeEntered(code: String)
    fun onDestroy()
    fun onEditAccountClicked()
    fun onResendEmail()
    fun onUpgradeClicked(textViewText: String)
    fun onLazyLoginClicked()
    fun setLayoutFromApiSession()
    fun setTheme(context: Context)
}